File: Presentations/PEPM2020/projectile2/java/Projectile/InputFormat.java
package Projectile;

import java.io.File;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.util.Scanner;

public class InputFormat {
    
    public static Object[] get_input(String filename) throws Exception {
        double v_launch;
        double theta;
        double p_target;
        
        PrintWriter outfile;
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.println("function get_input called with inputs: {");
        outfile.print("  filename = ");
        outfile.println(filename);
        outfile.println("  }");
        outfile.close();
        
        Scanner infile;
        infile = new Scanner(new File(filename));
        infile.nextLine();
        v_launch = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'v_launch' assigned to ");
        outfile.print(v_launch);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        theta = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'theta' assigned to ");
        outfile.print(theta);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        p_target = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'p_target' assigned to ");
        outfile.print(p_target);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.close();
        
        Object[] outputs = new Object[3];
        outputs[0] = v_launch;
        outputs[1] = theta;
        outputs[2] = p_target;
        return outputs;
    }
}

File: code/stable/truss/src/python/InputFormat.py
## \file InputFormat.py
# \author Ting-Yu Wu
# \brief Provides the function for reading inputs
## \brief Reads input from a file with the given file name
# \param filename name of the input file
# \param inParams structure holding the input values
def get_input(filename, inParams):
    outfile = open("log.txt", "a")
    print("function get_input called with inputs: {", file=outfile)
    print("  filename = ", end="", file=outfile)
    print(filename, end="", file=outfile)
    print(", ", file=outfile)
    print("  inParams = ", end="", file=outfile)
    print("Instance of InputParameters object", file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    infile = open(filename, "r")
    infile.readline()
    inParams.F_vect_1 = float(infile.readline())
    outfile = open("log.txt", "a")
    print("var 'inParams.F_vect_1' assigned ", end="", file=outfile)
    print(inParams.F_vect_1, end="", file=outfile)
    print(" in module InputFormat", file=outfile)
    outfile.close()
    infile.readline()
    inParams.x_1 = float(infile.readline())
    outfile = open("log.txt", "a")
    print("var 'inParams.x_1' assigned ", end="", file=outfile)
    print(inParams.x_1, end="", file=outfile)
    print(" in module InputFormat", file=outfile)
    outfile.close()
    infile.readline()
    inParams.x_2 = float(infile.readline())
    outfile = open("log.txt", "a")
    print("var 'inParams.x_2' assigned ", end="", file=outfile)
    print(inParams.x_2, end="", file=outfile)
    print(" in module InputFormat", file=outfile)
    outfile.close()
    infile.readline()
    inParams.θ_1 = float(infile.readline())
    outfile = open("log.txt", "a")
    print("var 'inParams.θ_1' assigned ", end="", file=outfile)
    print(inParams.θ_1, end="", file=outfile)
    print(" in module InputFormat", file=outfile)
    outfile.close()
    infile.readline()
    inParams.θ_2 = float(infile.readline())
    outfile = open("log.txt", "a")
    print("var 'inParams.θ_2' assigned ", end="", file=outfile)
    print(inParams.θ_2, end="", file=outfile)
    print(" in module InputFormat", file=outfile)
    outfile.close()
    infile.close()
